Types/Table.php

```diff
@@ -62,4 +65,52 @@ public static function alias(array $tokens) {
         if ($operation === SqlOperations::UPDATE) return $tokens['UPDATE'][0]['alias']['name'];
         return $tokens['FROM'][0]['alias']['name'];
     }
+
+    /**
+     * replaces the table in the tokens array with the given table
+     *
+     * @param  array $tokens
+     * @param  array $newTable
+     * @return array
+     */
+    public static function replace(array $tokens, $newTable) {
+        Assertions::assertHashMap('tokens', $tokens);
+
+        $operation = SqlOperation::create($tokens);
+        if ($operation === SqlOperations::INSERT) return self::replaceTableName($tokens, $newTable, 'INSERT', 1);
+        return $operation === SqlOperations::UPDATE ? self::replaceTableName($tokens, $newTable, 'UPDATE', 0) : self::replaceTableName($tokens, $newTable, 'FROM', 0);
+    }
+
+    /**
+     * replaces the table name with the related annotation
+     *
+     * @param  array        $tokens
+     * @param  RoutingTable $annotations
+     * @return array
+     */
+    public static function replaceWithAnnotation(array $tokens, RoutingTable $annotations = null) {
+        if (empty($annotations)) return $tokens;
+
+        $table        = Table::create($tokens);
+        $method       = HttpMethods::ofSqlOperation(SqlOperation::create($tokens));
+        $id           = Id::create($tokens);
+        $methodToCall = $method === HttpMethods::GET && empty($id) ? $method . 'All' : $method;
+
+        return !empty($annotations) && $annotations->get($table) !== null && $annotations->get($table)->$methodToCall() !== null ? Table::replace($tokens, $annotations->get($table)->$methodToCall()) : $tokens;
```

Comment:
Wir sind beide behindert :D Ich muss einfach nur fragen ob der instance of ist. Ich erwarte MaybeRoutingTable und sicher mich gegen das Maybe ab. Und wie macht man das? :D Gut, dass du das gesehen hast - was ein Quatsch!